### PR TITLE
Monorepo (WIP)

### LINKED ISSUE #138: Convert to monorepo
Convert the repo to a monorepo like Babel and use the [lerna](https://lernajs.io/) tool to help manage publishing of each of the modules.

### LINKED ISSUE #163: Looking for a new name!
Apollo Server is growing up, and we're looking for a new name that stands on its own. Ideas so far:

- graphql-server
- Pioneer
- Neptune

Ideas are most welcome. We'll probably pick a new name soon, so don't wait until you have the perfect name, just let us know your ideas. Extra points if the package name is still available on npm.

### PR DESCRIPTION
**Pull request by DxCx**
TODO:

- [x] fix node 4 on travis (@DxCx)
- [X] test versus private repository (@DxCx)
- [X] Renaming apollo-server to graphql-server. (#163)
- [x] Need to Create GitHub labels and modify lerna.json according it. (@helfer)
- [ ] Add Different README.md for each package (@helfer)
- [x] General Polish (@helfer)
- [ ] Review & Approve Documentation (@helfer)
- [ ] Pending new lerna version with root-dev patch. (lerna/lerna#357)
- [ ] Update CHANGELOG.md with your change (include reference to issue & this PR)
- [ ] Rebase your changes on master so that they can be merged easily
- [X] Make sure all tests and linter rules pass

Closes #138 
Closed #163

### CODE REVIEW

**Comment by DxCx** [comment]
Hi @nnance (@helfer pointed at you), @stubailo 
I've done intial work for monorepo, still have small stuff to close. (TODO on PR)
but would like to get your first feedback

**Comment by nnance** [comment]
@DxCx sorry I just now got the time to look this over.  I will give feedback as soon as I finish an initial review.

**Comment by DxCx** [comment]
Just 'npm install' ;)
The mono repo init is a postinstall script

**Comment by DxCx** [comment]
ok guys, i'm done with my items. everything seems to work against private npm repo.
waiting for more feedback now and the reset of todo items

File: packages/apollo-server-graphiql/tsconfig.json [hunk 1]
```diff
@@ -8,9 +8,15 @@
     "noImplicitAny": false,
     "rootDir": "./src",
     "outDir": "./dist",
-    "allowSyntheticDefaultImports": true,
+    "allowSyntheticDefaultImports": false,
```
Comment (nnance) [inline packages/apollo-server-graphiql/tsconfig.json]: why change this to false?

**Comment by nnance** [comment]
@DxCx @helfer I have confirmed that with a couple of pretty simple changes we can now remove the use babel from the project.  Should we include those changes in this PR or a new PR based on this branch?

File: lerna.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
+{
+  "lerna": "2.0.0-beta.30",
+  "version": "0.3.2",
+  "changelog": {
+    "repo": "apollostack/apollo-server",
+    "labels": {
```
Comment (DxCx) [inline lerna.json]: They are github issue/PR labels which @helfer needs to create then define there.
Later on it is used for creating changelog between versions

File: packages/apollo-server-graphiql/tsconfig.json (hunk 1 above)
Comment (DxCx) [inline packages/apollo-server-graphiql/tsconfig.json]: Because its creating arbitrary ///<reference ... in generated d.ts

**Comment by DxCx** [comment]
Yeah that was my next goal,
I wanted this to be merged (but yet to be released) then i planned on removing babel.
If you want you can take it but i think it should be done step by step..

File: lerna.json [hunk 3]
```diff
@@ -0,0 +1,16 @@
+{
+  "lerna": "2.0.0-beta.30",
```
Comment (DxCx) [inline lerna.json]: I think that Im using features that seems to require version 2..
Looks like the rest of the big projects that uses lerna are using beta as well

**Comment by DxCx** [comment]
@nnance regarding atom errors
Maybe it parses package.json and not node_modules?
Please make sure that it's there on node_modules

It is peerDependancy for packages that ises it, and global devDep for all the packages
(Want to make sure there is a single version installed)

File: lerna.json (hunk 2 above)
Comment (nnance) [inline lerna.json]: @DxCx  I might be able to create the labels.  Is creating a changelog a feature of lerna?

**Comment by stubailo** [comment]
Are we still blocked on something for this PR?

**Comment by DxCx** [comment]
yeah @stubailo i'm updaing the TODOs every time status changes.
right now as you can see we need some of helfer's attention :)

**Comment by helfer** [comment]
@stubailo or @nnance can one take over the items in the todo list currently assigned to me?

**Pull request merged**